site/pages/disruption-detail/[disruptionId].page.tsx

```diff
@@ -331,7 +331,8 @@ const DisruptionDetail = ({
         });
     };
 
-    const nextIndex = getLargestConsequenceIndex(disruption) + 1;
+    const nextIndex =
+        disruption.consequences && disruption.consequences.length > 0 ? getLargestConsequenceIndex(disruption) + 1 : 0;
```

Comment:
Is this check needed? It's also done within the `getLargestConsequenceIndex` function

Reply:
Also it'll have the side effect of overwriting deleted consequences

Reply:
Without this check, If we delete a consequence from a disruption so no consequence exists then create a new consequence the index will be 1 not 0 which I think is incorrect?

I've seen this cause a bug when creating a disruption from a template - can show you on a call.

Reply:
Historically that was intentional so that the consequence index would always be unique within a disruption so it wouldn't end up overwriting previous consequences. That might not be an issue anymore with how edits are handled  now though. Happy to jump on a call to see the bug it's causing.

Reply:
reverted it back to original code